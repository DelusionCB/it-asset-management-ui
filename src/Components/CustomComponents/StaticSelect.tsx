import * as React from 'react';
import {Col, FormGroup, Input, Label} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'
import {staticSelectPropTypes} from '../../Types/types.customComponents';

function StaticSelect ({label, validation, value, onChange, options, id, placeholder, disabled = false, invalid = false}: staticSelectPropTypes): JSX.Element {
    const {t} = useTranslation()

    function getOptions (options: string[]): JSX.Element[] {
        return options.map((value, index) => {
            return (
                <option key={index} value={value}>{t(`select.${value}`)}</option>
            )
        })
    }

    return (
        <Col className='textfield'>
            <FormGroup>
                <Label for={id}>
                    {t(label)}
                </Label>
                <Input
                    id={id}
                    disabled={disabled}
                    invalid={invalid}
                    aria-invalid={invalid}
                    name={id}
                    placeholder={`${t(placeholder)}`}
                    type='select'
                    value={value}
                    onChange={(e) => { onChange(e); }}
                >
                    {getOptions(options)}
                </Input>
            </FormGroup>
        </Col>
    );
}

export default StaticSelect;
