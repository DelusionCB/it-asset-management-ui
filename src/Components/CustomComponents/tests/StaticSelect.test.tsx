import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import StaticSelect from '../StaticSelect';
import '@testing-library/jest-dom'

describe('<StaticSelect />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <StaticSelect
                invalid={false}
                id='id'
                label='label'
                onChange={jest.fn()}
                placeholder={'placeholder'}
                value={undefined}
                disabled={false}
                options={['1', '2']}
            />,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
