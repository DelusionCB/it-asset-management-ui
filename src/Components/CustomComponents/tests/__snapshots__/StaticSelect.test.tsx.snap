// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticSelect /> Snapshot renders correctly 1`] = `
<div
  className="textfield col"
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="id"
    >
      label
    </label>
    <select
      aria-invalid={false}
      className="form-select"
      disabled={false}
      id="id"
      name="id"
      onChange={[Function]}
      placeholder="placeholder"
    >
      <option
        value="1"
      >
        select.1
      </option>
      <option
        value="2"
      >
        select.2
      </option>
    </select>
  </div>
</div>
`;
